### PR TITLE
wasm blacklist and whitelist

### PR DESCRIPTION
**Pull request by xzav3r**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/wasm/keeper/proposal_handler.go [hunk 1]
```diff
@@ -236,3 +248,30 @@ func handleUpdateInstantiateConfigProposal(ctx sdk.Context, k types.ContractOpsK
...
 	return nil
 }
+
+func handleUpdateDeploymentWhitelistProposal(ctx sdk.Context, k types.ContractOpsKeeper, p types.UpdateDeploymentWhitelistProposal) error {
+	if err := p.ValidateBasic(); err != nil {
+		return err
+	}
+
+	var config types.AccessConfig
+	if len(p.DistributorAddrs) == 0 {
```
Comment (BananaLF) [inline x/wasm/keeper/proposal_handler.go:263]: need clear config.address when AccessTypeNobody or AccessTypeEverybody

File: x/wasm/types/proposal_custom.pb.go [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+}
+
+func (m *UpdateWASMContractMethodBlockedListProposal) GetIsDelete() bool {
+	if m != nil {
+		return m.IsDelete
+	}
+	return false
+}
+
+type ContractMethods struct {
```
Comment (BananaLF) [inline x/wasm/types/proposal_custom.pb.go:143]: need change name

File: x/wasm/types/proposal_custom.go [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+	}
+	return c.Name + " " + c.Extra
+}
+
+func (c *ContractMethods) DeleteMethods(methods []*Method) {
+	for _, method := range methods {
+		for i := range c.Methods {
+			if c.Methods[i].FullName() == method.FullName() {
+				//delete method
+				c.Methods = append(c.Methods[:i], c.Methods[i+1:]...)
```
Comment (BananaLF) [inline x/wasm/types/proposal_custom.go:122]: add break much better

File: x/wasm/keeper/keeper.go [hunk 4]
```diff
@@ -195,6 +197,80 @@ func (k Keeper) GetStoreKey() sdk.StoreKey {
...
+}
+
+func (k Keeper) getContractMethodBlockedList(ctx sdk.Context, contractAddr string) *types.ContractMethods {
+	store := k.ada.NewStore(ctx.GasMeter(), ctx.KVStore(k.storeKey), nil)
+	key := types.GetContractMethodBlockedListPrefix(contractAddr)
+	data := store.Get(key)
+	var blockedMethods types.ContractMethods
+	err := proto.Unmarshal(data, &blockedMethods)
+	if err != nil {
+		k.Logger(ctx).Error("getContractMethodBlockedList", "unmarshal error", err)
```
Comment (BananaLF) [inline x/wasm/keeper/keeper.go]: I think unmarshal err must panic. if len == 0 can return empty obj

File: x/wasm/keeper/keeper.go [hunk 5]
```diff
@@ -195,6 +197,80 @@ func (k Keeper) GetStoreKey() sdk.StoreKey {
...
+	store := k.ada.NewStore(ctx.GasMeter(), ctx.KVStore(k.storeKey), nil)
+	it := sdk.KVStorePrefixIterator(store, types.GetContractMethodBlockedListPrefix(""))
+	defer it.Close()
+
+	var cms []*types.ContractMethods
+	for ; it.Valid(); it.Next() {
+		var method types.ContractMethods
+		err := proto.Unmarshal(it.Value(), &method)
+		if err != nil {
+			k.Logger(ctx).Error("getAllBlockedList", "unmarshal error", err)
```
Comment (BananaLF) [inline x/wasm/keeper/keeper.go]: the same as  getContractMethodBlockedList

File: x/wasm/keeper/proposal_handler.go (hunk 1 above)
Comment (xzav3r) [reply x/wasm/keeper/proposal_handler.go:263]: config.Address is "" by default

File: x/wasm/types/proposal_custom.go (hunk 3 above)
Comment (xzav3r) [reply x/wasm/types/proposal_custom.go:122]: good idea

File: x/wasm/keeper/keeper.go (hunk 4 above)
Comment (xzav3r) [reply x/wasm/keeper/keeper.go]: panic is ok.
proto.Unmarshal() will return nil when len(data) == 0

**Review by BananaLF — APPROVED** [review]

**Pull request merged**